crates/sparrow-merge/src/in_memory_batches.rs

```diff
@@ -53,7 +55,9 @@ impl InMemoryBatches {
             let (version, old) = &*write;
             let version = *version;
 
-            let merged = if old.num_rows() == 0 {
+            let merged = if !self.retained {
+                old.clone()
```

Comment:
This keeps the empty batch forever. We could probably put this in a struct to allow updating just the version (rather than using a tuple. Or we could modify the fields. @jordanrfrazier for thoughts on if it is worth doing to clean thsi up.

Reply:
Probably worth a comment at least